profiler/src/ProfilerEngine/Datadog.Profiler.Native.Linux/LibrariesInfoCache.cpp

```diff
@@ -0,0 +1,111 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+
+#include "LibrariesInfoCache.h"
+
+#include "Log.h"
+
+#include <string.h>
+
+LibrariesInfoCache* LibrariesInfoCache::s_instance = nullptr;
+
+// This function is exposed by the ld_preload wrapper library.
+// This allows us to know if we have to reload the cache or not
+extern "C" unsigned long long dd_nb_calls_to_dlopen_dlclose() __attribute__((weak));
```

Comment:
IMO we should really use well-defined types like `uint64_t` where we can

Reply:
good point, I think I went there because I did realize that `uint64_t` was available in C

Reply:
Oh, right, if we use uint64_t here then we can't have the same declaration on C side. Never mind then